### PR TITLE
ci: Selective triggering

### PR DESCRIPTION
**Pull request by ko3n1g**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: .github/workflows/cicd-main.yml [hunk 1]
```diff
@@ -35,40 +35,37 @@ jobs:
...
-    if: ${{ github.event.label.name == 'Run CICD' }}
+    if: ${{ github.event.label.name == 'Run CICD' || github.event_name == 'workflow_dispatch' }}
     steps:
     - name: Clean server from old files
       run: |
-        docker container prune --filter "until=24h" --force
-        docker image prune -a --filter "until=24h" --force
-
-
-  cicd-test-container-setup:
```
Comment (pablo-garay) [inline .github/workflows/cicd-main.yml:67]: There was a reason to keep cicd-cluster-clean separate from cicd-test-container-setup . So let's please get the old form back? :)

Comment (ko3n1g) [reply .github/workflows/cicd-main.yml:67]: Sure let me change that back! Out of curiosity, what's the reason to have it separated?

Comment (pablo-garay) [reply .github/workflows/cicd-main.yml:67]: Not sure i remember right now out of the top of my mind. Will need to get back to this a bit later

Comment (pablo-garay) [reply .github/workflows/cicd-main.yml:67]: Also, better to focus this PR strictly to the changes for the selective trigger purpose :)

Comment (ko3n1g) [reply .github/workflows/cicd-main.yml:67]: sure, i readded it

File: .github/workflows/cicd-main.yml [hunk 2]
```diff
@@ -115,8 +120,9 @@ jobs:
         ### \'\'
   
   L0_Unit_Tests_GPU:
-    needs: [cicd-test-container-setup]
+    needs: [cicd-test-container-setup, pre-flight]
```
Comment (pablo-garay) [inline .github/workflows/cicd-main.yml:123]: Thinking whether we need to do this
needs: [cicd-test-container-setup, pre-flight]
for each test or it's better to just make
cicd-test-container-setup depend on pre-flight
and not have all the rest of tests depend on pre-flight

**Review by pablo-garay — CHANGES_REQUESTED** [review]

**Review by pablo-garay — APPROVED** [review]

**Comment by pablo-garay** [comment]
Example of triggering a single test: https://github.com/NVIDIA/NeMo/actions/runs/10618944819/job/29435552797
Example of triggering all: https://github.com/NVIDIA/NeMo/actions/runs/10619071709/job/29435808466

**Pull request merged**